src/hooks/transactions/useEstimateFee.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { useCallback } from 'react';
+import { useApiCallsContext } from '../../contexts/ApiCallsContextProvider';
+import { useSourceTarget } from '../../contexts/SourceTargetContextProvider';
+import { TransactionActionCreators } from '../../actions/transactionActions';
+import { useUpdateTransactionContext, useTransactionContext } from '../../contexts/TransactionContext';
+import { getSubstrateDynamicNames } from '../../util/getSubstrateDynamicNames';
+import useLaneId from '../chain/useLaneId';
+
+export const useEstimateFee = () => {
+  const { createType, stateCall } = useApiCallsContext();
+  const laneId = useLaneId();
+  const {
+    sourceChainDetails: { chain: sourceChain },
+    targetChainDetails: { chain: targetChain }
+  } = useSourceTarget();
+  const { dispatchTransaction } = useUpdateTransactionContext();
+  const { estimatedFeeMethodName } = getSubstrateDynamicNames(targetChain);
+  const { payload } = useTransactionContext();
```

Comment:
I think it would be much cleaner to have all parameters passed as an argument to `getEstimateFeeCall`. Now on the component level you don't really know when to update `useEffect` and it will quickly get messy.

Basically IMHO this hook should only be responsible for calling the asynchronous method, but should not depend on any existing state.

Reply:
https://github.com/paritytech/parity-bridges-ui/pull/186#discussion_r650903055

Reply:
well in a way it is reusable because it is being used not only in Transfer component but in Remark and CustomCall.
Anyway , i have set the payload as input parameter as you recommended.